### PR TITLE
GetSize Bug Fix

### PR DESCRIPTION
**Pull request by chatrapathi06**
## Description

Fixes #1 GetSize support for the local image

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| Linux   |    ✅   |
| iOS     |    ❌     |
| Android |    ❌     |
| Windows |   ❌     |

## Checklist

* [x] I have tested this on a device/simulator for each compatible OS
* [ ] I added the documentation in `README.md`
* [ ] I updated the typings files (`privateTypes.ts`, `types.ts`)
* [ ] I added a sample use of the API (`example/App.js`)

### CODE REVIEW

File: ReactSkia/core_modules/RSkResourceManager.h [hunk 1]
```diff
@@ -0,0 +1,4 @@
+//resource manager Macro.
+#define ASSETSPATH  "./assets/config/custom/assets/"
+#define EXTENTION  ".png"
+#define GENERATEPATH(URI)  ASSETSPATH+URI+EXTENTION
```
Comment (munezbn) [inline ReactSkia/core_modules/RSkResourceManager.h]: I dont think you need to use + for concatenating in pre-processor

File: ReactSkia/components/RSkComponentImage.cpp [hunk 2]
```diff
@@ -155,8 +159,15 @@ sk_sp<SkImage> RSkComponentImage::getLocalImageData(string sourceUri) {
...
 
 inline string RSkComponentImage::generateUriPath(string path) {
-  if(path.substr(0, 14) == "file://assets/")
+  if(path.substr(0, 14) == "file://assets/"){
     path = "./" + path.substr(7);
+  } else if(path.substr(0,4) == "http"){
+
+  } else {
+
+    path =  GENERATEPATH(path);
```
Comment (munezbn) [inline ReactSkia/components/RSkComponentImage.cpp]: This else condition may not be right if the path is something like "file://xxxxx"

File: ReactSkia/components/RSkComponentImage.cpp [hunk 3]
```diff
@@ -184,7 +195,7 @@ RnsShell::LayerInvalidateMask RSkComponentImage::updateComponentProps(SharedProp
         //TODO - need to send the onEnd event to APP if it is abort.
         isRequestInProgress_=false;
       }
-      networkImageData_.reset();
+      networkImageData_ = nullptr;
```
Comment (munezbn) [inline ReactSkia/components/RSkComponentImage.cpp]: Any particular reason for this change ?

File: ReactSkia/core_modules/RSkImageLoader.cpp [hunk 4]
```diff
@@ -117,6 +110,40 @@ void RSkImageLoaderModule::getImageSize(std::string uri, CxxModule::Callback res
     RNS_LOG_DEBUG("Send Request to network");
     sharedCurlNetworking->sendRequest(remoteCurlRequest,folly::dynamic::object());
     imageRequestList_.insert(std::pair<std::string, std::shared_ptr<CurlRequest> >(uri,remoteCurlRequest));
+  }else if(std::filesystem::exists(path)|| (uri.substr(0, 14) == "file://assets/")){ // Check if ION PATH exits.
```
Comment (munezbn) [inline ReactSkia/core_modules/RSkImageLoader.cpp]: comment must be Check if assets path exists. Also, can we have a telephonic review on this part ?

File: ReactSkia/components/RSkComponentImage.cpp (hunk 3 above)
Comment (chatrapathi06) [reply ReactSkia/components/RSkComponentImage.cpp]: NO, clean up issue

File: ReactSkia/core_modules/RSkResourceManager.h (hunk 1 above)
Comment (chatrapathi06) [reply ReactSkia/core_modules/RSkResourceManager.h]: here URI is string type and AssestPath and Extension is char * type. 
So, I did that. 
while pre-processing  it replace in  "./assets/config/custom/assets/" +uri+".png"
so, it not leads to error.

File: ReactSkia/components/RSkComponentImage.cpp [hunk 5]
```diff
@@ -48,7 +49,7 @@ void RSkComponentImage::OnPaint(SkCanvas *canvas) {
     imageData = RSkImageCacheManager::getImageCacheManagerInstance()->findImageDataInCache(imageProps.sources[0].uri.c_str());
     if(imageData) break;
 
-    if (imageProps.sources[0].type == ImageSource::Type::Local) {
+    if (imageProps.sources[0].type == ImageSource::Type::Local || imageProps.sources[0].uri.substr(0,4) != "http") {
```
Comment (chatrapathi06) [inline ReactSkia/components/RSkComponentImage.cpp]: check what type we are getting.

File: ReactSkia/components/RSkComponentImage.cpp (hunk 2 above)
Comment (chatrapathi06) [reply ReactSkia/components/RSkComponentImage.cpp]: check file path, can we give as file:: in application.
use http:// || https:// if check

File: ReactSkia/core_modules/RSkImageLoader.cpp [hunk 6]
```diff
@@ -65,17 +67,8 @@ std::string RSkImageLoaderModule::getName() {
...
-    handleRejectBlock(rejectBlock);
-    return;
-  }
-  imageData = RSkImageCacheManager::getImageCacheManagerInstance()->findImageDataInCache(uri.c_str());
-  if(imageData) {
-    handleResolveBlock(resolveBlock,imageData);
-    return;
-  } else { /*getting image data from network */
+ auto path = GENERATEPATH(uri); 
+ if(uri.substr(0,4) == "http"){ /*getting image data from network */
```
Comment (chatrapathi06) [inline ReactSkia/core_modules/RSkImageLoader.cpp]: follow the above comment.

File: ReactSkia/core_modules/RSkImageLoader.cpp [hunk 7]
```diff
@@ -117,6 +110,40 @@ void RSkImageLoaderModule::getImageSize(std::string uri, CxxModule::Callback res
...
+    decodedimageCacheData imageCacheData;
+    if(!path.c_str()) {
+      RNS_LOG_ERROR("Invalid File");
+      handleRejectBlock(rejectBlock);
+      return;
+    }
+    data = SkData::MakeFromFileName(path.c_str());
+    if (!data) {
+      RNS_LOG_ERROR("Unable to make SkData for path : " << path.c_str());
+      handleRejectBlock(rejectBlock);
```
Comment (chatrapathi06) [inline ReactSkia/core_modules/RSkImageLoader.cpp:131]: return is missing

File: ReactSkia/components/RSkComponentImage.cpp (hunk 5 above)
Comment (chatrapathi06) [reply ReactSkia/components/RSkComponentImage.cpp]: Source type getting it as Remote.

File: ReactSkia/components/RSkComponentImage.cpp [hunk 8]
```diff
@@ -155,8 +168,17 @@ sk_sp<SkImage> RSkComponentImage::getLocalImageData(string sourceUri) {
 }
 
 inline string RSkComponentImage::generateUriPath(string path) {
-  if(path.substr(0, 14) == "file://assets/")
+  if((path.substr(0,4) == "http://")||(path.substr(0,4) == "https://"))
```
Comment (shefayaparwin) [inline ReactSkia/components/RSkComponentImage.cpp]: is it (0,4) for substr ??
is  in't it- (0,7)/(0/8) ??

File: ReactSkia/core_modules/RSkImageLoader.cpp [hunk 9]
```diff
@@ -65,17 +67,12 @@ std::string RSkImageLoaderModule::getName() {
...
-  }
-  imageData = RSkImageCacheManager::getImageCacheManagerInstance()->findImageDataInCache(uri.c_str());
-  if(imageData) {
-    handleResolveBlock(resolveBlock,imageData);
-    return;
-  } else { /*getting image data from network */
+  sk_sp<SkData> data;
+  decodedimageCacheData imageCacheData;
+
+  if((uri.substr(0,7) == "http://") || (uri.substr(0,8) == "https://")){
```
Comment (shefayaparwin) [inline ReactSkia/core_modules/RSkImageLoader.cpp]: As it defined in more than one place, can it be a macro ??

**Review by shefayaparwin — APPROVED** [review]

File: ReactSkia/sdk/RNSAssetManager.h [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+namespace rns {
+namespace sdk {
+
+class RNSAssetManager{
+private:
+  RNSAssetManager();
+  static RNSAssetManager *RNSAssetManagerInstance_;
+  static std::mutex mutex_;
+  folly::dynamic assetDataFile_ = folly::dynamic::object;
+  fstream assetFile_;
```
Comment (munezbn) [inline ReactSkia/sdk/RNSAssetManager.h]: We dont need this ?

File: ReactSkia/sdk/RNSAssetManager.cpp [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    std::lock_guard<std::mutex> lock(mutex_);
+  if (!RNSAssetManagerInstance_) {
+    RNSAssetManagerInstance_ = new RNSAssetManager();
+  }
+  return RNSAssetManagerInstance_;
+}
+
+string RNSAssetManager::getAssetPath(string uri){
+  std::string result="";
+  if(!assetFile_.is_open()){
```
Comment (munezbn) [inline ReactSkia/sdk/RNSAssetManager.cpp]: Instead of this check use , if assetDataFile_["assets"] key exists

**Review by munezbn — APPROVED** [review]

**Pull request merged**